etcdserver/etcdserverpb/rpc.proto

```diff
@@ -510,6 +510,9 @@ message Compare {
     // value is the value of the given key, in bytes.
     bytes value = 7;
   }
+  // range_end compares over the range [key, range_end). See RangeRequest.
+  bytes range_end = 8;
```

Comment:
when range_end is given, all the keys within the range will be compared against the same target? Probably we should make the comment more clear?

Also can you give me an example use case of this?